### PR TITLE
Fix TopN dimension extractions being applied twice

### PR DESCRIPTION
**Pull request by drcrallen**

### CODE REVIEW

File: processing/src/main/java/io/druid/query/topn/TopNQueryQueryToolChest.java [hunk 1]
```diff
@@ -284,8 +281,7 @@ public String apply(@Nullable AggregatorFactory input)
                       values.put(name, fn.manipulate(aggregatorFactories[i], input.getMetric(name)));
                     }
 
-                    final Object dimValue = input.getDimensionValue(dimension);
-                    values.put(dimension, extractionFn == null ? dimValue : extractionFn.apply(dimValue));
+                    values.put(dimension, input.getDimensionValue(dimension));
```
Comment (fjy) [inline processing/src/main/java/io/druid/query/topn/TopNQueryQueryToolChest.java]: is the extractionFn being applied in this call?

Comment (drcrallen) [inline processing/src/main/java/io/druid/query/topn/TopNQueryQueryToolChest.java]: No, not here. it is either being applied at a deeper (column select) or higher (post merge manipulator) level than here.

**Comment by drcrallen** [comment]
Turns out all the tests did silly thing like "extract first character" or "retain missing values" which can be applied as many times as it wants.

**Comment by xvrl** [comment]
do we need a test for groupBy as well?

**Comment by drcrallen** [comment]
@xvrl No turns out it is covered, I'll put in a PR to make extra sure though.

**Comment by fjy** [comment]
@drcrallen any indication of performance improvement?

**Comment by drcrallen** [comment]
@fjy For what part?

**Comment by fjy** [comment]
@drcrallen by not applying the extraction fn multiple times, is there a performance improvement?

**Comment by drcrallen** [comment]
@fjy Probably, but it should only be for dimension extraction. I do not have numbers for it.

**Comment by fjy** [comment]
:+1:

**Pull request merged**